app/Http/Controllers/GuestController.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Article;
+use Illuminate\Http\Request;
+
+class GuestController extends Controller
+{
+    public function home()
+    {
+        return view('guest.home')
+            ->with('articles', Article::with('category')->orderby('id', 'desc')->limit(3)->get());
+    }
+
+    public function show(Article $article)
+    {
+        return view('guest.show')
+            ->with('article', Article::with('tags', 'category')->findOrFail($article->id));
```

Comment:
If using route model binding why search for article second time? Two basically same queries here. If you want to eager load just use `load` https://laravel.com/docs/8.x/eloquent-relationships#lazy-eager-loading I see you have installed `barryvdh/laravel-debugbar` package. Didn't it showed it?
```php
$article->load(''tags', 'category');
```

Reply:
sorry, i didn't know that i can lazy eager load it. Yes, i saw in the laravel debugbar that there are two identical queries being queried and i tried to optimize it and remove the duplicate. At first i tried to $article->with('tags', 'category')->get() but it does not retrieve the tag and categories of the article that the route model binding retrieved but rather retrieves all of the articles with tags and categories. And then i tried to $article->with('tags', 'category')->first() but it just retrieves the first article with its tags and categories. So i just kinda sticked with it cause my initial solution work even though it duplicates the query. 

But now i realized that there is a load method thanks to you. I will definitely keep this one in mind.